src/pybind/mgr/telemetry/module.py

```diff
@@ -386,10 +386,9 @@ def gather_device_report(self):
                 rep['host_id'] = anon_host
 
             # anonymize device id
-            (vendor, model, serial) = devid.split('_')
             anon_devid = self.get_store('devid-id/%s' % devid)
             if not anon_devid:
-                anon_devid = '%s_%s_%s' % (vendor, model, uuid.uuid1())
+                anon_devid = f"{devid.rsplit('_', 1)[0]}_{uuid.uuid1()}"
```

Comment:
please note, this might fail the build on el7. as rpmbuild may precompile `*.py` files using `__python` which on el7 and el8 is python2.

Reply:
Thanks for noticing this!
I think the original fix [1] works on python2, but I guess we want to use the new syntax from now on. 
What would be the right way of fixing this?

[1] https://github.com/ceph/ceph/pull/32492#discussion_r363160050

Reply:
@yaarith sorry that i didn't realize #32492 was supposed to be backported to nautilus which is not python3 yet. i think the the right way is to 

1. use the original fix and
1. add a "Conflicts:" section in the commit message to explain the reason why we need to edit the commit.  because, in nautilus, ceph-mgr is still using python2, we should use python2 compatible syntax in this backport commit.

Reply:
Thanks for the guidance! I also didn't realize it was supposed to be backported when I opened it :-) 
Opened a new PR (https://github.com/ceph/ceph/pull/32948), couldn't fix this one...